consensus/polybft/fsm.go

```diff
@@ -262,6 +262,11 @@ func (f *fsm) Validate(proposal []byte) error {
 		return err
 	}
 
+	if err := extra.ValidateParentSignatures(block.Number(), f.polybftBackend, nil, f.parent, parentExtra,
```

Comment:
I this `ValidateParentSignatures` you are using the `Checkpoint` before validating it. Then, in `validateExtraData` which is called after this function, you are validating checkpoint data based on parent extra. I would say, move this nil checks for parent and current extra Checkpoint to that `ValidateParentSignatures` and remove it from the `validateExtraData`.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1208/commits/e1b8a65f095fb8167da7c8befe45153449b3b997.

However, I didn't put it to `ValidateParentSignatures`, as it is not using `Checkpoint` instance.
Btw I had doubts about whether to remove entirely check about `CheckpointData` being `nil` since we don't expect such use case ever to happen.